tests/sentry/api/endpoints/test_organization_repositories.py

```diff
@@ -35,6 +35,28 @@ def test_simple(self):
         assert response.status_code == 200, response.content
         assert len(response.data) == 1
         assert response.data[0]['id'] == six.text_type(repo.id)
+        assert response.data[0]['externalId'] is None
+
+    def test_get_integration_repository(self):
+        repo = Repository.objects.create(
+            name='getsentry/example',
+            organization_id=self.org.id,
+            external_id=12345,
+            provider='dummy',
+            config={'name': 'getsentry/example'}
+        )
+
+        response = self.client.get(self.url, format='json')
+
+        assert response.status_code == 200, response.content
+        assert len(response.data) == 1
+        first_row = response.data[0]
+        assert first_row['id'] == six.text_type(repo.id)
+        assert first_row['provider'] == {
+            'id': 'dummy',
+            'name': 'Example'
+        }
+        assert first_row['externalSlug'] == six.text_type(repo.external_id)
```

Comment:
Since part of the issue was that the id could be a number could you add a test with a numerical/alpha-numerical id?

Reply:
This test is checking that the `externalSlug` field matches `repo.external_id` which is a number-y string. Is that what you were thinking?

Reply:
@markstory I'm sorry; so I think I actually meant to put this on the JS tests! If I understood everything ok in the js tests, you only test: `externalSlug: 'example/repo-name',` Let me know if this isn't the case. And I'm ok with approving.